src/System.Runtime.Extensions/src/System/Environment.cs

```diff
@@ -0,0 +1,217 @@
+
+        public static void Exit(int exitCode) => EnvironmentAugments.Exit(exitCode);
+
+        public static int ExitCode { get { return EnvironmentAugments.ExitCode; } set { EnvironmentAugments.ExitCode = value; } }
+
+        public static void FailFast(string message) => FailFast(message, exception: null);
+
+        public static void FailFast(string message, Exception exception) => EnvironmentAugments.FailFast(message, exception);
+
+        public static string ExpandEnvironmentVariables(string name)
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if (name.Length == 0)
+            {
+                return name;
+            }
+
+            return ExpandEnvironmentVariablesCore(name);
+        }
+
+        public static string[] GetCommandLineArgs() => EnvironmentAugments.GetCommandLineArgs();
+
+        public static string GetEnvironmentVariable(string variable) => GetEnvironmentVariable(variable, EnvironmentVariableTarget.Process);
+
+        public static string GetEnvironmentVariable(string variable, EnvironmentVariableTarget target) => EnvironmentAugments.GetEnvironmentVariable(variable, target);
+
+        public static IDictionary GetEnvironmentVariables() => GetEnvironmentVariables(EnvironmentVariableTarget.Process);
+
+        public static IDictionary GetEnvironmentVariables(EnvironmentVariableTarget target) => EnvironmentAugments.GetEnvironmentVariables(target);
+
+        public static string GetFolderPath(SpecialFolder folder) => GetFolderPath(folder, SpecialFolderOption.None);
+
+        public static string GetFolderPath(SpecialFolder folder, SpecialFolderOption option)
+        {
+            if (!Enum.IsDefined(typeof(SpecialFolder), folder))
+            {
+                throw new ArgumentOutOfRangeException(nameof(folder), folder, SR.Format(SR.Arg_EnumIllegalVal, folder));
+            }
+
+            if (option != SpecialFolderOption.None && !Enum.IsDefined(typeof(SpecialFolderOption), option))
+            {
+                throw new ArgumentOutOfRangeException(nameof(option), option, SR.Format(SR.Arg_EnumIllegalVal, option));
+            }
+
+            return GetFolderPathCore(folder, option);
+        }
+
+        public static bool HasShutdownStarted => EnvironmentAugments.HasShutdownStarted;
+
+        public static bool Is64BitProcess => IntPtr.Size == 8;
+
+        public static bool Is64BitOperatingSystem => Is64BitProcess || Is64BitOperatingSystemWhen32BitProcess;
+
+        public static void SetEnvironmentVariable(string variable, string value) => SetEnvironmentVariable(variable, value, EnvironmentVariableTarget.Process);
+
+        public static void SetEnvironmentVariable(string variable, string value, EnvironmentVariableTarget target) => EnvironmentAugments.SetEnvironmentVariable(variable, value, target);
```

Comment:
Yes, I meant environment variables.

CoreCLR does not need to observe any changes made by corefx. CoreCLR initialization from environment is basically done during startup, any changes in environment after startup are ignored, and the runtime itself does not have a need to change environment variables.

The cache that avoids races and leaks on Unix should move without problems. It exists just to support BCL.